### PR TITLE
755 test-derive_advs_params

### PR DESCRIPTION
**Pull request by millerg23**
Put all unit tests related to derive_advs_params into unit test file test-derive_advs_params. Also added new tests, and implemented Unit Test Guidance with titles, and using sections/sub-sections to find unit tests easily.

### CODE REVIEW

**Comment by thomas-neitmann** [comment]
@millerg23 There appear to be some failing test cases. Please take a look an correct.

**Comment by thomas-neitmann** [comment]
Also please merge the `devel` branch into this one. There's currently a merge conflict that needs to be resolved.

**Comment by bms63** [comment]
@millerg23 Hey Gordon.  I'm happy to finish the review for this so we can close it out.  Let me know if this is still an active PR

File: tests/testthat/test-derive_advs_params.R [hunk 1]
```diff
@@ -0,0 +1,936 @@
...
+      by_vars = vars(USUBJID, VISIT),
+      method = "Mosteller",
+      get_unit_expr = VSSTRESU
+    ),
+    paste("It is expected that 'WEIGHT' is measured in 'kg'.\nIn the",
+          "input dataset it is measured in 'g'.")
+  )
+})
+
+test_that(paste("derive_param_bsa Test 01.03: BSA parameter NOT added to input dataset",
```
Comment (bms63) [inline tests/testthat/test-derive_advs_params.R:425]: What about removing the paste and just doing #nolint at the end of the line?

**Review by bms63 — APPROVED** [review]
Hi Gordon.  I made a quick suggestion around the paste function in your tests.  Happy to just wait on formatting the tests and do a more unified update to all the tests. #714

**Comment by bms63** [comment]
Oh I think we need to merge in devel and fix merge conflicts.

**Comment by bms63** [comment]
@millerg23 I think this one is ready to go if you can just fix merge conflicts.

**Comment by millerg23** [comment]
Thanks @bms63 will do this first thing tomorrow.

**Review by bms63 — APPROVED** [review]

**Comment by bms63** [comment]
@millerg23 Good to merge in?

**Pull request merged**